packages/url/src/test/index.js

```diff
@@ -273,6 +273,21 @@ describe( 'getFilename', () => {
 		expect( getFilename( 'image.jpg' ) ).toBe( 'image.jpg' );
 	} );
 
+	it( 'returns the filename for a variety of formats', () => {
+		expect( getFilename( 'https://wordpress.org/file.pdf' ) ).toBe(
+			'file.pdf'
+		);
+		expect(
+			getFilename( 'https://wordpress.org/image.webp?query=test' )
+		).toBe( 'image.webp' );
+		expect( getFilename( 'https://wordpress.org/video.mov#anchor' ) ).toBe(
+			'video.mov'
+		);
+		expect(
+			getFilename( 'http://localhost:8080/a/path/to/audio.mp3' )
+		).toBe( 'audio.mp3' );
+	} );
+
```

Comment:
It feels like we could make a single test by combining this one with `returns the filename part of the URL` and using `it.each()` to run over a variety of formats. What do you think?

Reply:
I wouldn't be opposed to that, since it could add some extra durability to this feature in the case of future changes.

Reply:
Sounds good!

Reply:
Do we need `it.each()` in this case? Rather than just combining them? 

https://github.com/WordPress/gutenberg/blob/d8bc25dddbf23e05bc2a0b4339a13789ec11777c/packages/url/src/test/index.js#L257-L286

Reply:
Strictly speaking, we don't _need_ to use `each()` here, but this is precisely the sort of scenario that it was designed for, and can make the intent of the test much clearer.

```js
it.each([
  [ 'https://wordpress.org/image.jpg', 'image.jpg' ],
  [ 'https://wordpress.org/image.jpg?query=test', 'image.jpg' ],
  [ 'https://wordpress.org/image.jpg#anchor', 'image.jpg' ],
  [ 'http://localhost:8080/a/path/to/an/image.jpg', 'image.jpg' ],
  [ '/path/to/an/image.jpg', 'image.jpg' ],
  [ 'path/to/an/image.jpg', 'image.jpg' ],
  [ '/image.jpg', 'image.jpg' ],
  [ 'https://wordpress.org/file.pdf', 'file.pdf' ],
  [ 'https://wordpress.org/image.webp?query=test', 'image.webp' ],
  [ 'https://wordpress.org/video.mov#anchor', 'video.mov' ],
  [ 'http://localhost:8080/a/path/to/audio.mp3', 'audio.mp3' ],
])( 'returns the filename part of the URL: %s', ( url, filename ) => { 
  expect( getFilename( url ) ).toBe( filename ); 
} ); 
```

Rather than having to read through the entire test to understand what's expected, we can see quite quickly that the intention is to have the same expectation for each set of parameters.

If we need to test additional parameters, it is clearer how to do that. And if we need to change the expectation, we will have more certainty that we've correctly applied that in every case.

Reply:
Thank you, @andrewhayward 🙌 This is very helpful!